Build/testing-docker/docker-compose.yml

```diff
@@ -1,6 +1,8 @@
 # copied from EXT:enetcache
 #  https://github.com/lolli42/enetcache/blob/main/Build/testing-docker/docker-compose.yml
----
+
+version: "3"
```

Comment:
I once asked Randy Fay @rfay about the sense of the version and he said it's not useful and could be removed.
Did you just copy it from somewhere or has it a meaning for you?

Reply:
@DavidBruchmann The runTests.sh script (and corresponding Build/testing-docker/lcoal/docker-compose.yml) was initially introduced in the TYPO3 core and then later used in extensions as well. Over time it has been continuously updated and changed. **Since TYPO3 v11, the docker-compose.yml script is no longer used in the core** (but runTests.sh still is but has changed). docker-compose is no longer used, docker or podman is used instead (depending on what is available on the system).

Also, docker-compose was phased out, should use "docker compose" (if using compose at all) or preferably have a switch to use what is available, e.g.

```
DOCKER_COMPOSE_COMMAND="docker-compose"
which docker-compose 2>/dev/null >/dev/null
if [ $? -ne 0 ];then
     DOCKER_COMPOSE_COMMAND="docker compose"
fi

```
So for this reason, I think the discussion is moot, it is better to rework the script based on newer version of runTests.sh from the core.

Compare newer version of runTests.sh:

* 10.4: https://github.com/TYPO3/typo3/blob/10.4/Build/Scripts/runTests.sh
* main: https://github.com/TYPO3/typo3/blob/main/Build/Scripts/runTests.sh


---

about version:

If you look in the TYPO3 10.4 core docker-compose.yml (https://github.com/TYPO3/typo3/blob/10.4/Build/testing-docker/local/docker-compose.yml) you can also see the version (but version 2.3):

`version: '2.3'`


It is my understanding that adding the version won't hurt, but omitting might cause problems with older docker-compose locally - where you can't control what contributors will be using. (If you run the pipeline only on GitHub should not be a problem). I am using docker-compose 1.25.3.

Officially, it seems "version" has been phased out.

Reply:
Thanks for the detailed explanation, @sypets, making it much clearer!